### PR TITLE
Preonic iso keymap

### PR DESCRIPTION
**Pull request by itsnoteasy**
iso uk keymap for preonic

changed the layout to match iso uk keycaps and changed function keys to 4x3 grid

## Description

modified the default preonic keymap for iso layout, and rearranged the function keys in a 4x3 grid.
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x ] I have added tests to cover my changes.
- [ x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/preonic/keymaps/isouk/keymap.c [hunk 1]
```diff
@@ -0,0 +1,385 @@
...
+            wait_ms(10);
+            SEND_STRING("00a7");
+            wait_ms(10);
+            tap_code(KC_ENT);
+      }
+            return false;
+         break;
+      case BEST:
+        if (record->event.pressed) {
+            tap_code16(LALT(KC_D));
```
Comment (drashna) [inline keyboards/preonic/keymaps/isouk/keymap.c:190]: `SEND_STRING_DELAY` will work much better for this.  It adds the delays between the key presses.

File: keyboards/preonic/keymaps/isouk/keymap.c [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+        muse_tempo-=1;
+      }
+    }
+  } else {
+    if (clockwise) {
+      register_code(KC_PGDN);
+      unregister_code(KC_PGDN);
+    } else {
+      register_code(KC_PGUP);
+      unregister_code(KC_PGUP);
```
Comment (drashna) [inline keyboards/preonic/keymaps/isouk/keymap.c:332]:
```suggestion
      tap_code(KC_PGDN);
    } else {
      tap_code(KC_PGUP);
```

File: keyboards/preonic/keymaps/isouk/keymap.c [hunk 3]
```diff
@@ -0,0 +1,385 @@
...
+    }
+  }
+}
+
+void dip_switch_update_user(uint8_t index, bool active) {
```
Comment (drashna) [inline keyboards/preonic/keymaps/isouk/keymap.c:337]:
```suggestion
bool dip_switch_update_user(uint8_t index, bool active) {
```

File: keyboards/preonic/keymaps/isouk/keymap.c [hunk 4]
```diff
@@ -0,0 +1,385 @@
...
+            }
+            break;
+        case 1:
+            if (active) {
+                muse_mode = true;
+            } else {
+                muse_mode = false;
+            }
+    }
+}
```
Comment (drashna) [inline keyboards/preonic/keymaps/isouk/keymap.c:353]:
```suggestion
    return false;
}
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/preonic/keymaps/isouk/keymap.c (hunk 1 above)
Comment (itsnoteasy) [reply keyboards/preonic/keymaps/isouk/keymap.c:190]: yes, but this works already and has no bugs also i don't have the skill to implement new commands. it's in a rarely run macro that's not going to slow down the keyboard.

**Review by drashna — APPROVED** [review]

File: keyboards/preonic/keymaps/isouk/keymap.c [hunk 5]
```diff
@@ -0,0 +1,384 @@
...
+      }
+    }
+  } else {
+    if (clockwise) {
+      tap_code(KC_PGDN);
+    } else {
+      tap_code(KC_PGUP);
+    }
+  }
+}
```
Comment (drashna) [inline keyboards/preonic/keymaps/isouk/keymap.c:333]:
```suggestion
    return false;
}
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/preonic/keymaps/isouk/keymap.c [hunk 6]
```diff
@@ -0,0 +1,385 @@
...
+  _______, RESET,   DEBUG,   _______, _______, _______, _______, TERM_ON, TERM_OFF,_______, _______, KC_DEL,  \
+  _______, _______, MU_MOD,  AU_ON,   AU_OFF,  AG_NORM, AG_SWAP, QWERTY,  COLEMAK, DVORAK,  _______, _______, \
+  _______, MUV_DE,  MUV_IN,  MU_ON,   MU_OFF,  MI_ON,   MI_OFF,  _______, _______, _______, _______, _______, \
+  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______  \
+)
+
+
+};
+
+
```
Comment (fauxpark) [inline keyboards/preonic/keymaps/isouk/keymap.c:173]:
```suggestion
[_ADJUST] = LAYOUT_preonic_grid(
  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,
  _______, QK_BOOT, DB_TOGG, _______, _______, _______, _______, _______, _______, _______, _______, KC_DEL,
  _______, _______, MU_MOD,  AU_ON,   AU_OFF,  AG_NORM, AG_SWAP, QWERTY,  COLEMAK, DVORAK,  _______, _______,
  _______, MUV_DE,  MUV_IN,  MU_ON,   MU_OFF,  MI_ON,   MI_OFF,  _______, _______, _______, _______, _______,
  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______
)
};
```

File: keyboards/preonic/keymaps/isouk/keymap.c [hunk 7]
```diff
@@ -0,0 +1,385 @@
...
+     * |------+------+------+------+------+------+------+------+------+------+------+------|
+     * |      |      |      |      |      |             |      | Next | Vol- | Vol+ | Play |
+     * `-----------------------------------------------------------------------------------'
+     */
+[_RAISE] = LAYOUT_preonic_grid( \
+        RELOAD,        BEST,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_SLSH,       LSFT(KC_8),    KC_LBRC, KC_RBRC, KC_MINS, \
+        KC_GRV,        KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_NO,   KC_NO,   LCTL(KC_MINS), LCTL(KC_PLUS), KC_NO,   KC_NO,   KC_INS, \
+        KC_NLCK,       KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_NO,   KC_NO,   KC_MINS,       KC_PLUS,       KC_NO,   KC_NO,   KC_BSLS, \
+        LSFT(KC_NUBS), KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_NO,   KC_NO,   KC_NO,         KC_PGUP,       KC_PGDN, KC_NO,   KC_NUHS, \
+        KC_NO,         KC_NO,   KC_NO,   KC_LGUI, KC_TRNS, KC_NO,   KC_NO,   KC_TRNS,       KC_MNXT,       KC_VOLD, KC_VOLU, KC_MPLY \
```
Comment (fauxpark) [inline keyboards/preonic/keymaps/isouk/keymap.c:146]:
```suggestion
[_RAISE] = LAYOUT_preonic_grid(
        RELOAD,        BEST,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_SLSH,       LSFT(KC_8),    KC_LBRC, KC_RBRC, KC_MINS,
        KC_GRV,        KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_NO,   KC_NO,   LCTL(KC_MINS), LCTL(KC_PLUS), KC_NO,   KC_NO,   KC_INS,
        KC_NLCK,       KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_NO,   KC_NO,   KC_MINS,       KC_PLUS,       KC_NO,   KC_NO,   KC_BSLS,
        LSFT(KC_NUBS), KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_NO,   KC_NO,   KC_NO,         KC_PGUP,       KC_PGDN, KC_NO,   KC_NUHS,
        KC_NO,         KC_NO,   KC_NO,   KC_LGUI, KC_TRNS, KC_NO,   KC_NO,   KC_TRNS,       KC_MNXT,       KC_VOLD, KC_VOLU, KC_MPLY
```

File: keyboards/preonic/keymaps/isouk/keymap.c [hunk 8]
```diff
@@ -0,0 +1,385 @@
...
+     * |------+------+------+------+------+------+------+------+------+------+------+------|
+     * |      |      |      |      |      |             |      | Next | Vol- | Vol+ | Play |
+     * `-----------------------------------------------------------------------------------'
+     */
+[_LOWER] = LAYOUT_preonic_grid( \
+        RELOAD,  BEST,    SECTION, KC_HASH, LALT(KC_F4),  KC_PERC, KC_CIRC, KC_AMPR, KC_ASTR, LSFT(KC_LBRC), LSFT(KC_RBRC), KC_PSCR, \
+        KC_TILD, KC_F1,   KC_F2,   KC_F3,   KC_F4,        KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,         KC_NO,         KC_NO, \
+        KC_CAPS, KC_F5,   KC_F6,   KC_F7,   KC_F8,        KC_NO,   KC_NO,   KC_UNDS, KC_EQL,  KC_NO,         KC_NO,         KC_NO, \
+        KC_NUBS, KC_F9,   KC_F10,  KC_F11,  KC_F12,       KC_NO,   KC_NUHS, KC_NUBS, KC_HOME, KC_END,        KC_NO,         LSFT(KC_NUHS), \
+        KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_TRNS,      KC_NO,   KC_NO,   KC_TRNS, KC_MNXT, KC_VOLD,       KC_VOLU,       KC_MPLY \
```
Comment (fauxpark) [inline keyboards/preonic/keymaps/isouk/keymap.c:125]:
```suggestiob
[_LOWER] = LAYOUT_preonic_grid(
        RELOAD,  BEST,    SECTION, KC_HASH, LALT(KC_F4),  KC_PERC, KC_CIRC, KC_AMPR, KC_ASTR, LSFT(KC_LBRC), LSFT(KC_RBRC), KC_PSCR,
        KC_TILD, KC_F1,   KC_F2,   KC_F3,   KC_F4,        KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,         KC_NO,         KC_NO,
        KC_CAPS, KC_F5,   KC_F6,   KC_F7,   KC_F8,        KC_NO,   KC_NO,   KC_UNDS, KC_EQL,  KC_NO,         KC_NO,         KC_NO,
        KC_NUBS, KC_F9,   KC_F10,  KC_F11,  KC_F12,       KC_NO,   KC_NUHS, KC_NUBS, KC_HOME, KC_END,        KC_NO,         LSFT(KC_NUHS),
        KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_TRNS,      KC_NO,   KC_NO,   KC_TRNS, KC_MNXT, KC_VOLD,       KC_VOLU,       KC_MPLY
```

File: keyboards/preonic/keymaps/isouk/keymap.c [hunk 9]
```diff
@@ -0,0 +1,385 @@
...
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * | Brite| Ctrl | Alt  | GUI  |Lower |    Space    |Raise | Left | Down |  Up  |Right |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_DVORAK] = LAYOUT_preonic_grid( \
+  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_BSPC, \
+  KC_TAB,  KC_QUOT, KC_COMM, KC_DOT,  KC_P,    KC_Y,    KC_F,    KC_G,    KC_C,    KC_R,    KC_L,    KC_DEL,  \
+  KC_ESC,  KC_A,    KC_O,    KC_E,    KC_U,    KC_I,    KC_D,    KC_H,    KC_T,    KC_N,    KC_S,    KC_SLSH, \
+  KC_LSFT, KC_SCLN, KC_Q,    KC_J,    KC_K,    KC_X,    KC_B,    KC_M,    KC_W,    KC_V,    KC_Z,    KC_ENT,  \
+  BACKLIT, KC_LCTL, KC_LALT, KC_LGUI, LOWER,   KC_SPC,  KC_SPC,  RAISE,   KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT  \
```
Comment (fauxpark) [inline keyboards/preonic/keymaps/isouk/keymap.c:104]:
```suggestion
[_DVORAK] = LAYOUT_preonic_grid(
  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_BSPC,
  KC_TAB,  KC_QUOT, KC_COMM, KC_DOT,  KC_P,    KC_Y,    KC_F,    KC_G,    KC_C,    KC_R,    KC_L,    KC_DEL,
  KC_ESC,  KC_A,    KC_O,    KC_E,    KC_U,    KC_I,    KC_D,    KC_H,    KC_T,    KC_N,    KC_S,    KC_SLSH,
  KC_LSFT, KC_SCLN, KC_Q,    KC_J,    KC_K,    KC_X,    KC_B,    KC_M,    KC_W,    KC_V,    KC_Z,    KC_ENT,
  BACKLIT, KC_LCTL, KC_LALT, KC_LGUI, LOWER,   KC_SPC,  KC_SPC,  RAISE,   KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT
```

File: keyboards/preonic/keymaps/isouk/keymap.c [hunk 10]
```diff
@@ -0,0 +1,385 @@
...
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * | Brite| Ctrl | Alt  | GUI  |Lower |    Space    |Raise | Left | Down |  Up  |Right |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_COLEMAK] = LAYOUT_preonic_grid( \
+  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_BSPC, \
+  KC_TAB,  KC_Q,    KC_W,    KC_F,    KC_P,    KC_G,    KC_J,    KC_L,    KC_U,    KC_Y,    KC_SCLN, KC_DEL,  \
+  KC_ESC,  KC_A,    KC_R,    KC_S,    KC_T,    KC_D,    KC_H,    KC_N,    KC_E,    KC_I,    KC_O,    KC_QUOT, \
+  KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_K,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_ENT,  \
+  BACKLIT, KC_LCTL, KC_LALT, KC_LGUI, LOWER,   KC_SPC,  KC_SPC,  RAISE,   KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT  \
```
Comment (fauxpark) [inline keyboards/preonic/keymaps/isouk/keymap.c:83]:
```suggestion
[_COLEMAK] = LAYOUT_preonic_grid(
  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_BSPC,
  KC_TAB,  KC_Q,    KC_W,    KC_F,    KC_P,    KC_G,    KC_J,    KC_L,    KC_U,    KC_Y,    KC_SCLN, KC_DEL,
  KC_ESC,  KC_A,    KC_R,    KC_S,    KC_T,    KC_D,    KC_H,    KC_N,    KC_E,    KC_I,    KC_O,    KC_QUOT,
  KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_K,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_ENT,
  BACKLIT, KC_LCTL, KC_LALT, KC_LGUI, LOWER,   KC_SPC,  KC_SPC,  RAISE,   KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT
```

File: keyboards/preonic/keymaps/isouk/keymap.c [hunk 11]
```diff
@@ -0,0 +1,385 @@
...
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * | ralt | Ctrl | Alt  | GUI  |Lower |    Space    |Raise | Left | Down |  Up  |Right |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_QWERTY] = LAYOUT_preonic_grid( \
+  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_BSPC, \
+  KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_DEL,  \
+  KC_ESC,  KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, \
+  KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_ENT,  \
+  KC_RALT, KC_LCTL, KC_LALT, KC_TRNS, LOWER,   KC_SPC,  KC_SPC,  RAISE,   KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT  \
```
Comment (fauxpark) [inline keyboards/preonic/keymaps/isouk/keymap.c:62]:
```suggestion
[_QWERTY] = LAYOUT_preonic_grid(
  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_BSPC,
  KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_DEL,
  KC_ESC,  KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,
  KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_ENT,
  KC_RALT, KC_LCTL, KC_LALT, KC_TRNS, LOWER,   KC_SPC,  KC_SPC,  RAISE,   KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT
```

**Review by fauxpark — CHANGES_REQUESTED** [review]
This keymap seems pretty specialised to your use case, so I think the folder would be better named after your GitHub username.

File: keyboards/preonic/keymaps/isouk/keymap.c (hunk 6 above)
Comment (itsnoteasy) [reply keyboards/preonic/keymaps/isouk/keymap.c:173]: uh, that's a matter of opinion. the keymap is more or less stock preonic with the addition of a few macros. there's not going to be many users other than myself. i'd like to leave the name as is to save the bother of making changes.

File: keyboards/preonic/keymaps/isouk/keymap.c (hunk 8 above)
Comment (itsnoteasy) [reply keyboards/preonic/keymaps/isouk/keymap.c:125]: Is there a mistake here? i don't see any changes.

Comment (fauxpark) [reply keyboards/preonic/keymaps/isouk/keymap.c:125]: Fixed.

**Pull request closed without merging**